### PR TITLE
Add torchfix linter

### PR DESCRIPTION
**Pull request by titu1994**
# What does this PR do ?

Adds the torchfix linter to all PRs 

**Collection**: [Core]

# Changelog 
- Scans the codebase python files to analyse torch fixes / improvements

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

### CODE REVIEW

**Review by ericharper — APPROVED** [review]
LGTM. Thanks!!

**Review by XuesongYang — COMMENTED** [review]
LGTM for TTS related changes. Thanks!

**nit**: it seems more elegant and clear if we could move further by using explicit vector norm or matrix norm, such as using `torch.linalg.vector_norm` or `torch.linalg.matrix_norm`. 

**Question or Potential bugs**: regarding `torch.norm(A)` --> `torch.linalg.norm(A)`, the docstring said "If dim= None and ord != None, A must be 1D or 2D.". I made a simple check, see below, not sure if our test will fail when addressing 3+ dimension tensors.
```
In [7]: a
Out[7]: tensor([-4., -3., -2., -1.,  0.,  1.,  2.,  3.])

In [8]: b = a.reshape(4,2)

In [9]: b
Out[9]:
tensor([[-4., -3.],
        [-2., -1.],
        [ 0.,  1.],
        [ 2.,  3.]])

In [10]: c = a.reshape(2,2,2)

In [11]: c
Out[11]:
tensor([[[-4., -3.],
         [-2., -1.]],

        [[ 0.,  1.],
         [ 2.,  3.]]])

In [12]: torch.norm(a)
Out[12]: tensor(6.6332)

In [13]: torch.norm(b)
Out[13]: tensor(6.6332)

In [14]: torch.norm(c)
Out[14]: tensor(6.6332)

In [15]: torch.linalg.norm(a)
Out[15]: tensor(6.6332)

In [16]: torch.linalg.norm(b)
Out[16]: tensor(6.6332)

In [17]: torch.linalg.norm(c)
Out[17]: tensor(6.6332)

In [18]: torch.linalg.norm(c, ord=2)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[18], line 1
----> 1 torch.linalg.norm(c, ord=2)

RuntimeError: linalg.norm: If dim is not specified but ord is, the input must be 1D or 2D. Got 3D.

In [19]: torch.norm(c, p=2)
Out[19]: tensor(6.6332)
```

File: nemo/collections/tts/modules/common.py [hunk 1]
```diff
@@ -192,7 +192,7 @@ def __init__(self, c):
         # Ensure determinant is 1.0 not -1.0
         if torch.det(W) < 0:
             W[:, 0] = -1 * W[:, 0]
-        p, lower, upper = torch.lu_unpack(*torch.lu(W))
+        p, lower, upper = torch.lu_unpack(*torch.linalg.lu(W))
```
Comment (XuesongYang) [inline nemo/collections/tts/modules/common.py:195]: this change is **not correct**. Per the statement in pytorch 2.0 doc, "[torch.linalg.lu()](https://pytorch.org/docs/stable/generated/torch.linalg.lu.html#torch.linalg.lu) computes the LU decomposition with partial pivoting of a possibly non-square matrix. It is a composition of [lu_factor()](https://pytorch.org/docs/stable/generated/torch.linalg.lu_factor.html#torch.linalg.lu_factor) and [torch.lu_unpack()],(https://pytorch.org/docs/stable/generated/torch.lu_unpack.html#torch.lu_unpack).", while "LU, pivots, info = torch.lu(A, compute_pivots) should be replaced with `LU, pivots = torch.linalg.lu_factor(A, compute_pivots)`". So the correct change should be,
```suggestion
        p, lower, upper = torch.linalg.lu(W)
```

See my validation code snippet,
```
In [25]: a_lu
Out[25]:
tensor([[[-8.3251e-01, -2.0748e-01, -1.6207e-03],
         [-3.0119e-01,  2.1512e+00,  1.7865e-01],
         [-5.4838e-01,  9.3401e-02, -2.2064e-01]],

        [[ 8.0517e-01, -1.3429e+00,  6.1186e-01],
         [ 9.6715e-01,  2.8609e+00, -1.9533e-01],
         [-5.8811e-01, -3.4345e-01,  6.3169e-01]]])

In [26]: pivots
Out[26]:
tensor([[2, 2, 3],
        [2, 2, 3]], dtype=torch.int32)

In [27]: p, lower, upper = torch.lu_unpack(*torch.lu(a))

In [28]: p
Out[28]:
tensor([[[0., 1., 0.],
         [1., 0., 0.],
         [0., 0., 1.]],

        [[0., 1., 0.],
         [1., 0., 0.],
         [0., 0., 1.]]])

In [29]: lower
Out[29]:
tensor([[[ 1.0000,  0.0000,  0.0000],
         [-0.3012,  1.0000,  0.0000],
         [-0.5484,  0.0934,  1.0000]],

        [[ 1.0000,  0.0000,  0.0000],
         [ 0.9671,  1.0000,  0.0000],
         [-0.5881, -0.3434,  1.0000]]])

In [30]: upper
Out[30]:
tensor([[[-8.3251e-01, -2.0748e-01, -1.6207e-03],
         [ 0.0000e+00,  2.1512e+00,  1.7865e-01],
         [ 0.0000e+00,  0.0000e+00, -2.2064e-01]],

        [[ 8.0517e-01, -1.3429e+00,  6.1186e-01],
         [ 0.0000e+00,  2.8609e+00, -1.9533e-01],
         [ 0.0000e+00,  0.0000e+00,  6.3169e-01]]])

In [31]: p, lower, upper = torch.lu_unpack(*torch.linalg.lu(a))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[31], line 1
----> 1 p, lower, upper = torch.lu_unpack(*torch.linalg.lu(a))

TypeError: lu_unpack(): argument 'unpack_data' (position 3) must be bool, not Tensor

In [32]: p, lower, upper = torch.linalg.lu(a)

In [33]: p
Out[33]:
tensor([[[0., 1., 0.],
         [1., 0., 0.],
         [0., 0., 1.]],

        [[0., 1., 0.],
         [1., 0., 0.],
         [0., 0., 1.]]])

In [34]: lower
Out[34]:
tensor([[[ 1.0000,  0.0000,  0.0000],
         [-0.3012,  1.0000,  0.0000],
         [-0.5484,  0.0934,  1.0000]],

        [[ 1.0000,  0.0000,  0.0000],
         [ 0.9671,  1.0000,  0.0000],
         [-0.5881, -0.3434,  1.0000]]])

In [35]: upper
Out[35]:
tensor([[[-8.3251e-01, -2.0748e-01, -1.6207e-03],
         [ 0.0000e+00,  2.1512e+00,  1.7865e-01],
         [ 0.0000e+00,  0.0000e+00, -2.2064e-01]],

        [[ 8.0517e-01, -1.3429e+00,  6.1186e-01],
         [ 0.0000e+00,  2.8609e+00, -1.9533e-01],
         [ 0.0000e+00,  0.0000e+00,  6.3169e-01]]])
```

**Review by XuesongYang — CHANGES_REQUESTED** [review]

**Comment by kit1980** [comment]
Should this be revived?

**Pull request closed without merging**